src/common/xo/migrate-vm-modal/index.js

```diff
@@ -178,10 +183,16 @@ export default class MigrateVmModalBody extends BaseComponent {
       }
 
       this.setState({
+        defaultSr,
+        defaultSrConnectedToHost,
         doNotMigrateVdis,
         host,
         intraPool,
-        mapVdisSrs: doNotMigrateVdis ? undefined : mapValues(vdis, vdi => defaultSr),
+        mapVdisSrs: doNotMigrateVdis
+          ? undefined
+          : defaultSrConnectedToHost
+            ? mapValues(vdis, vdi => defaultSr)
```

Comment:
Avoid doing this: you should send a *main* `sr` to xo-server which it should use as fallback if no appropriate entry in `mapVdisSrs`.